[MacCatalyst] [Release] Inconsistent Behavior Between Debug and Release Builds

### Current behavior

We have encountered an issue with our Uno Platform application targeting Mac Catalyst, where the application behaves differently in the Release package compared to Debug Which makes the app unusable. 

**Primary Issue:** Data Display Failure in FeedView
Issue happens after a simple login (works as expected), when navigating to the Home page where we have a FeedView containing an ItemsRepeater but none is displayed

**Secondary Issue:** XAML Navigation Failure on Button Press
On the same Home page, attempt to use XAML-based navigation by pressing a button designed for this purpose. In the Release build, pressing the navigation button results in no action, contrary to the expected navigation behavior. 

This functionalities works as intended in the Debug build.

Currently targeting to .net8

### Expected behavior

The ItemsRepeater should dynamically load and display the data, with appropriate handling for None, Progress, or Error states through their respective templates and Button should navigte to the correct page.



### How to reproduce it (as minimally and precisely as possible)

We wont be able to provide a repro project on this case as the issue is only ocurring on the app installed by the release package

### Workaround

_No response_

### Works on UWP/WinUI

None

### Environment

_No response_

### NuGet package version(s)

```
<PackageVersion Include="CommunityToolkit.WinUI.UI.Controls" Version="7.1.2" />
<PackageVersion Include="CommunityToolkit.Mvvm" Version="8.2.2" />
<PackageVersion Include="Geolocation" Version="1.2.1" />
<PackageVersion Include="Microsoft.Extensions.Logging.Console" Version="8.0.0" />
<PackageVersion Include="Microsoft.Windows.SDK.BuildTools" Version="10.0.22621.2428" />
<PackageVersion Include="Microsoft.WindowsAppSDK" Version="1.4.231219000" />
<PackageVersion Include="Uno.Core.Extensions.Logging.Singleton" Version="4.0.1" />
<PackageVersion Include="Uno.Extensions.Authentication" Version="4.0.0" />
<PackageVersion Include="Uno.Extensions.Authentication.WinUI" Version="4.0.0" />

<PackageVersion Include="Uno.Extensions.Configuration" Version="4.0.0" />
<PackageVersion Include="Uno.Extensions.Hosting" Version="4.0.0" />
<PackageVersion Include="Uno.Extensions.Hosting.WinUI" Version="4.0.0" />
<PackageVersion Include="Uno.Extensions.Http" Version="4.0.0" />
<PackageVersion Include="Uno.Extensions.Http.WinUI" Version="4.0.0" />
<PackageVersion Include="Uno.Extensions.Http.Refit" Version="4.0.0" />
<PackageVersion Include="Uno.Extensions.Localization" Version="4.0.0" />
<PackageVersion Include="Uno.Extensions.Localization.WinUI" Version="4.0.0" />
<PackageVersion Include="Uno.Extensions.Logging.OSLog" Version="1.7.0" />
<PackageVersion Include="Uno.Extensions.Logging.WinUI" Version="4.0.0" />
<PackageVersion Include="Uno.Extensions.Navigation" Version="4.0.0" />
<PackageVersion Include="Uno.Extensions.Navigation.WinUI" Version="4.0.0" />
<PackageVersion Include="Uno.Extensions.Navigation.Toolkit.WinUI" Version="4.0.0" />
<PackageVersion Include="Uno.Extensions.Reactive" Version="4.0.0" />
<PackageVersion Include="Uno.Extensions.Reactive.WinUI" Version="4.0.0" />
<PackageVersion Include="Uno.Extensions.Serialization" Version="4.0.0" />
<PackageVersion Include="Uno.Extensions.Serialization.Http" Version="4.0.0" />
<PackageVersion Include="Uno.Extensions.Serialization.Refit" Version="4.0.0" />
<PackageVersion Include="Uno.Material.WinUI" Version="4.1.1" />
<PackageVersion Include="Uno.Dsp.Tasks" Version="1.2.8" />
<PackageVersion Include="Uno.Toolkit.WinUI" Version="5.1.5" />
<PackageVersion Include="Uno.Toolkit.WinUI.Material" Version="5.1.5" />
<PackageVersion Include="Uno.Resizetizer" Version="1.2.1" />

<PackageVersion Include="Uno.UI.Adapter.Microsoft.Extensions.Logging" Version="5.1.50" />
<PackageVersion Include="Uno.UniversalImageLoader" Version="1.9.36" />
<PackageVersion Include="Uno.WinUI" Version="5.1.50" />
<PackageVersion Include="Uno.WinUI.Lottie" Version="5.1.50"/>
<PackageVersion Include="Uno.WinUI.RemoteControl" Version="5.1.50" />
<PackageVersion Include="Xamarin.Google.Android.Material" Version="1.9.0.2" />
<PackageVersion Include="Mapsui.Uno.WinUI" Version="4.1.2" />
<PackageVersion Include="Uno.Extensions.Reactive.Messaging" Version="2.5.11" />
<PackageVersion Include="Uno.WinUI.Svg" Version="5.0.41" />
<PackageVersion Include="SkiaSharp.Views.Uno" Version="2.88.6" />
```

### Affected platforms

Mac Catalyst

### IDE

_No response_

### IDE version

_No response_

### Relevant plugins

_No response_

### Anything else we need to know?

Please let me know if I can provide any other info I missed

@jeromelaban After migrating to Uno.sdk we are facing this errors when building mac catalyst:

```
CSC : error CS1508: Resource identifier 'monotouch_content_Assets_fIcons_fguidllogo.png' has already been used in this assembly [/Users/maurococaro/GuidlAppCreator/Guidl_CMS/Guidl_CMS/Guidl_CMS/Guidl_CMS.csproj]
CSC : error CS1508: Resource identifier 'monotouch_content_Assets_fIcons_fheadphones.png' has already been used in this assembly [/Users/maurococaro/GuidlAppCreator/Guidl_CMS/Guidl_CMS/Guidl_CMS/Guidl_CMS.csproj]
CSC : error CS1508: Resource identifier 'monotouch_content_Assets_flogo.png' has already been used in this assembly [/Users/maurococaro/GuidlAppCreator/Guidl_CMS/Guidl_CMS/Guidl_CMS/Guidl_CMS.csproj]
CSC : error CS1508: Resource identifier 'monotouch_content_Assets_fsplashlogo.png' has already been used in this assembly [/Users/maurococaro/GuidlAppCreator/Guidl_CMS/Guidl_CMS/Guidl_CMS/Guidl_CMS.csproj]
CSC : error CS1508: Resource identifier 'monotouch_content_Assets_floading.gif' has already been used in this assembly [/Users/maurococaro/GuidlAppCreator/Guidl_CMS/Guidl_CMS/Guidl_CMS/Guidl_CMS.csproj]
CSC : error CS1508: Resource identifier 'monotouch_content_Assets_frecording.gif' has already been used in this assembly [/Users/maurococaro/GuidlAppCreator/Guidl_CMS/Guidl_CMS/Guidl_CMS/Guidl_CMS.csproj]
CSC : error CS1508: Resource identifier 'monotouch_content_Assets_floginimage.jpg' has already been used in this assembly [/Users/maurococaro/GuidlAppCreator/Guidl_CMS/Guidl_CMS/Guidl_CMS/Guidl_CMS.csproj]
CSC : error CS1508: Resource identifier '__monotouch_content_Assets_fsplash.jpg' has already been used in this assembly [/Users/maurococaro/GuidlAppCreator/Guidl_CMS/Guidl_CMS/Guidl_CMS/Guidl_CMS.csproj]
```
Binlog: [net8.0-maccatalyst-Debug-maccatalyst-x64.zip](https://github.com/unoplatform/uno/files/14272556/net8.0-maccatalyst-Debug-maccatalyst-x64.zip)

I've compared the .csproj and spolution items with a newly created app using Uno template wizard and I see no difference or anything that could be causing that errors

@jeromelaban Thank you, I'm getting the same error but the message changed:

```
CSC : error CS1508: Resource identifier '__monotouch_content_obj_fDebug_fnet8.0-maccatalyst_funoresizetizer_fr_fAssets_fIcons_fback.png' has already been used in this assembly [/Users/maurococaro/GuidlAppCreator/Guidl_CMS/Guidl_CMS/Guidl_CMS/Guidl_CMS.csproj]
CSC : error CS1508: Resource identifier '__monotouch_content_obj_fDebug_fnet8.0-maccatalyst_funoresizetizer_fr_fAssets_fIcons_fback.scale-100.png' has already been used in this assembly [/Users/maurococaro/GuidlAppCreator/Guidl_CMS/Guidl_CMS/Guidl_CMS/Guidl_CMS.csproj]
CSC : error CS1508: Resource identifier '__monotouch_content_obj_fDebug_fnet8.0-maccatalyst_funoresizetizer_fr_fAssets_fIcons_fback.scale-125.png' has already been used in this assembly [/Users/maurococaro/GuidlAppCreator/Guidl_CMS/Guidl_CMS/Guidl_CMS/Guidl_CMS.csproj]
CSC : error CS1508: Resource identifier '__monotouch_content_obj_fDebug_fnet8.0-maccatalyst_funoresizetizer_fr_fAssets_fIcons_fback.scale-150.png' has already been used in this assembly [/Users/maurococaro/GuidlAppCreator/Guidl_CMS/Guidl_CMS/Guidl_CMS/Guidl_CMS.csproj]
CSC : error CS1508: Resource identifier '__monotouch_content_obj_fDebug_fnet8.0-maccatalyst_funoresizetizer_fr_fAssets_fIcons_fback.scale-200.png' has already been used in this assembly [/Users/maurococaro/GuidlAppCreator/Guidl_CMS/Guidl_CMS/Guidl_CMS/Guidl_CMS.csproj]
CSC : error CS1508: Resource identifier '__monotouch_content_obj_fDebug_fnet8.0-maccatalyst_funoresizetizer_fr_fAssets_fIcons_fback.scale-300.png' has already been used in this assembly [/Users/maurococaro/GuidlAppCreator/Guidl_CMS/Guidl_CMS/Guidl_CMS/Guidl_CMS.csproj]
CSC : error CS1508: Resource identifier '__monotouch_content_obj_fDebug_fnet8.0-maccatalyst_funoresizetizer_fr_fAssets_fIcons_fback.scale-400.png' has already been used in this assembly [/Users/maurococaro/GuidlAppCreator/Guidl_CMS/Guidl_CMS/Guidl_CMS/Guidl_CMS.csproj]
```
Binlog:
[net8.0-maccatalyst-Debug-maccatalyst-x64 (2).zip](https://github.com/unoplatform/uno/files/14297398/net8.0-maccatalyst-Debug-maccatalyst-x64.2.zip)

--- a/src/Uno.Sdk/Sdk/Sdk.props
+++ b/src/Uno.Sdk/Sdk/Sdk.props
@@ -24,6 +24,9 @@ Copyright (C) Uno Platform Inc. All rights reserved.
 		<UsingUnoSdk>true</UsingUnoSdk>
 		<UnoVersion>DefaultUnoVersion</UnoVersion>
 
+		<AssetsFolder Condition=" '$(AssetsFolder)' == '' ">Assets\</AssetsFolder>
+		<AssetsFolder>$([MSBuild]::EnsureTrailingSlash('$(AssetsFolder)'))</AssetsFolder>
+
 		<_UnoSdkTargetsDirectory>$([MSBuild]::EnsureTrailingSlash($([MSBuild]::NormalizePath('$(MSBuildThisFileDirectory)..\targets'))))</_UnoSdkTargetsDirectory>
 		<CustomAfterDirectoryBuildProps>$(CustomAfterDirectoryBuildProps);$(_UnoSdkTargetsDirectory)Uno.Import.SolutionConfig.props</CustomAfterDirectoryBuildProps>
 		<CustomAfterDirectoryBuildProps>$(CustomAfterDirectoryBuildProps);$(_UnoSdkTargetsDirectory)Uno.IsPlatform.props</CustomAfterDirectoryBuildProps>

--- a/src/Uno.Sdk/targets/Uno.DefaultItems.targets
+++ b/src/Uno.Sdk/targets/Uno.DefaultItems.targets
@@ -5,24 +5,50 @@
 
 	<ItemGroup Condition=" !$(IsWinAppSdk) ">
 		<!-- Include all images by default - matches the __WindowsAppSdkDefaultImageIncludes property in the WindowsAppSDK -->
-		<Content Include="Assets\**;**/*.png;**/*.bmp;**/*.jpg;**/*.dds;**/*.tif;**/*.tga;**/*.gif" 
-			Exclude="$(DefaultItemExcludes);$(DefaultExcludesInProjectFolder);**\*.svg">
-			<Link>$([MSBuild]::MakeRelative ('$(MSBuildProjectDirectory)', '%(Identity)'))</Link>
-			<IsDefaultItem>true</IsDefaultItem>
-		</Content>
-		<Page Include="**\*.xaml" Exclude="$(DefaultItemExcludes);$(DefaultExcludesInProjectFolder)">
-			<Link>$([MSBuild]::MakeRelative ('$(MSBuildProjectDirectory)', '%(Identity)'))</Link>
-			<IsDefaultItem>true</IsDefaultItem>
-		</Page>
-		<Compile Update="**\*.xaml.cs">
-			<DependentUpon>%(Filename)</DependentUpon>
-			<Link>$([MSBuild]::MakeRelative ('$(MSBuildProjectDirectory)', '%(Identity)'))</Link>
-			<IsDefaultItem>true</IsDefaultItem>
-		</Compile>
-		<PRIResource Include="**\*.resw" Exclude="$(DefaultItemExcludes);$(DefaultExcludesInProjectFolder)">
-			<Link>$([MSBuild]::MakeRelative ('$(MSBuildProjectDirectory)', '%(Identity)'))</Link>
-			<IsDefaultItem>true</IsDefaultItem>
-		</PRIResource>
+		<Content Include="$(AssetsFolder)**"
+			Exclude="$(DefaultItemExcludes);$(DefaultExcludesInProjectFolder);**\*.svg;**\*.cs;@(Content)"
+			IsDefaultItem="true"
+			Link="$([MSBuild]::MakeRelative ('$(MSBuildProjectDirectory)', '%(Identity)'))" />
+		<Content Include="**\*.png"
+			Exclude="$(DefaultItemExcludes);$(DefaultExcludesInProjectFolder);@(Content)"
+			IsDefaultItem="true"
+			Link="$([MSBuild]::MakeRelative ('$(MSBuildProjectDirectory)', '%(Identity)'))" />
+		<Content Include="**\*.bmp"
+			Exclude="$(DefaultItemExcludes);$(DefaultExcludesInProjectFolder);@(Content)"
+			IsDefaultItem="true"
+			Link="$([MSBuild]::MakeRelative ('$(MSBuildProjectDirectory)', '%(Identity)'))" />
+		<Content Include="**\*.jpg"
+			Exclude="$(DefaultItemExcludes);$(DefaultExcludesInProjectFolder);@(Content)"
+			IsDefaultItem="true"
+			Link="$([MSBuild]::MakeRelative ('$(MSBuildProjectDirectory)', '%(Identity)'))" />
+		<Content Include="**\*.dds"
+			Exclude="$(DefaultItemExcludes);$(DefaultExcludesInProjectFolder);@(Content)"
+			IsDefaultItem="true"
+			Link="$([MSBuild]::MakeRelative ('$(MSBuildProjectDirectory)', '%(Identity)'))" />
+		<Content Include="**\*.tif"
+			Exclude="$(DefaultItemExcludes);$(DefaultExcludesInProjectFolder);@(Content)"
+			IsDefaultItem="true"
+			Link="$([MSBuild]::MakeRelative ('$(MSBuildProjectDirectory)', '%(Identity)'))" />
+		<Content Include="**\*.tga"
+			Exclude="$(DefaultItemExcludes);$(DefaultExcludesInProjectFolder);@(Content)"
+			IsDefaultItem="true"
+			Link="$([MSBuild]::MakeRelative ('$(MSBuildProjectDirectory)', '%(Identity)'))" />
+		<Content Include="**\*.gif"
+			Exclude="$(DefaultItemExcludes);$(DefaultExcludesInProjectFolder);@(Content)"
+			IsDefaultItem="true"
+			Link="$([MSBuild]::MakeRelative ('$(MSBuildProjectDirectory)', '%(Identity)'))" />
+		<Page Include="**\*.xaml"
+			Exclude="$(DefaultItemExcludes);$(DefaultExcludesInProjectFolder);@(Page)"
+			IsDefaultItem="true"
+			Link="$([MSBuild]::MakeRelative ('$(MSBuildProjectDirectory)', '%(Identity)'))" />
+		<Compile Update="**\*.xaml.cs"
+			DependentUpon="%(Filename)"
+			IsDefaultItem="true"
+			Link="$([MSBuild]::MakeRelative ('$(MSBuildProjectDirectory)', '%(Identity)'))" />
+		<PRIResource Include="**\*.resw"
+			Exclude="$(DefaultItemExcludes);$(DefaultExcludesInProjectFolder);@(PRIResource)"
+			IsDefaultItem="true"
+			Link="$([MSBuild]::MakeRelative ('$(MSBuildProjectDirectory)', '%(Identity)'))" />
 	</ItemGroup>
 
 	<ItemGroup Condition=" '$(PkgUno_Resizetizer)'!='' ">

